src/main/java/net/wurstclient/hacks/FullbrightHack.java

```diff
@@ -46,14 +49,23 @@ public FullbrightHack()
 		
 		EVENTS.add(UpdateListener.class, this);
 	}
-	
+
+	private static double gamma;
+
 	@Override
 	public void onUpdate()
 	{
+		if (!isEnabled()) {
+			if (MC.options.gamma <= 1 && smoothGamma == SmoothGamma.STOPPED) {
+				gamma = MC.options.gamma;
+			}
+		}
+
 		if(isEnabled() && method.getSelected() == Method.GAMMA)
 			approachGamma(16);
 		else
-			approachGamma(0.5);
+			if (MC.options.gamma > gamma)
+			approachGamma(gamma);

```

Comment:
...and then this will force the user's brightness to zero.

Reply:
this means that it will only approach gamma if the settings is bigger than the saved gamma, but iirc gamma value is updated before that line

Reply:
a check if MC.options.gamma is bigger than 1 can also be used, but I think that this wouldn't affect it.
And with onUpdate being triggered, the gamma value will be saved, if the fullbright is off